docs/topics/jvm/jvm-test-using-junit.md

```diff
@@ -1,190 +1,306 @@
+     This helps to avoid having to use nullable types (`TodoRepository?`) in your tests.
 
-## Add the code to test it
+4. Add a test inside the `TodoRepositoryTest` class, covering the initial repository state and its size:
 
-1. Open the `Main.kt` file in `src/main/kotlin`.
+   ```kotlin
+   @Test
+   @DisplayName("Should start with empty repository")
+   fun shouldStartEmpty() {
+       Assertions.assertEquals(0, repository.size())
+       Assertions.assertTrue(repository.all.isEmpty())
+   }
+   ```
 
-   The `src` directory contains Kotlin source files and resources. 
-   The `Main.kt` file contains sample code that prints `Hello, World!`.
+   * Compared to Java's static import, Jupiter's `Assertions` is imported as a class, and you can use it as a qualifier
+     for assertion functions.
+   * Instead of writing `.getAll()` cals, you can access Java getters as properties in Kotlin with `repository.all`. 
 
-2. Create the `Sample` class with the `sum()` function that adds two integers together:
+5. Verify copy behavior for all items:
 
    ```kotlin
-   class Sample() {
-       fun sum(a: Int, b: Int): Int {
-           return a + b
-       }
+   @Test
+   @DisplayName("Should return defensive copy of items")
+   fun shouldReturnDefensiveCopy() {
+       repository.add(testItem1)
+
+       val items1 = repository.all
+       val items2 = repository.all
+
+       Assertions.assertNotSame(items1, items2)
+       Assertions.assertThrows(
+           UnsupportedOperationException::class.java
+       ) { items1.clear() }
+       Assertions.assertEquals(1, repository.size())
    }
    ```
 
-## Create a test
-
-1. In IntelliJ IDEA, select **Code** | **Generate** | **Test...** for the `Sample` class:
-
-   ![Generate a test](generate-test.png)
+   * To get a Java class object from a Kotlin class, use `::class.java`.
+   * You can split complex assertions across multiple lines without needing special continuation characters.
 
-2. Specify the name of the test class. For example, `SampleTest`:
+6. Test finding items by ID:
 
-   ![Create a test](create-test.png)
+   ```kotlin
+   @Test
+   @DisplayName("Should find item by ID")
+   fun shouldFindItemById() {
+       repository.add(testItem1)
+       repository.add(testItem2)
 
-   IntelliJ IDEA creates the `SampleTest.kt` file in the `test` directory.
-   This directory contains Kotlin test source files and resources.
+        val found = repository.getById(testItem1.id())
 
-   > You can also manually create a `*.kt` file for tests in `src/test/kotlin`.
-   >
-   {style="note"}
+        Assertions.assertTrue(found.isPresent)
+        Assertions.assertEquals(testItem1, found.get())
+   }
+   ```
 
-3. Add the test code for the `sum()` function in `SampleTest.kt`:
+   Kotlin works smoothly with the Java `Optional` API, automatically converting getter methods to properties,
+   `isPresent()` method is accessed as a property.
```

Comment:
Can be a separate sentence: 
The `isPresent()` method can be accessed as a property.